Apple Silicon support

monojenkins opened the pull request:
!! This PR is a copy of mono/mono#20166,  please do not edit or review it in this repo !!<br/>Do not automatically approve this PR:<br/><br/>* Consider how the changes affect configurations in this repo,<br/>* Check effects on files that are not mirrored,<br/>* Identify test cases that may be needed in this repo.<br/><br/>!! Merge the PR only after the original PR is merged !!<br/><hr/><br/>
<!--
Thank you for your Pull Request!

If you are new to contributing to Mono, please try to do your best at conforming to our coding guidelines http://www.mono-project.com/community/contributing/coding-guidelines/ but don't worry if you get something wrong. One of the project members will help you to get things landed.

Does your pull request fix any of the existing issues? Please use the following format: Fixes #issue-number
-->

vargaz commented:
The mirror pr is failing but it cannot be fixed in the original PR since it requires changes to mono.proj.

CoffeeFlux commented:
Historically I've dealt with that by pushing to the mirror PR with additional changes once I'm sure the original is ready.

vargaz committed:
Apple Silicon support

<!--
Thank you for your Pull Request!

If you are new to contributing to Mono, please try to do your best at conforming to our coding guidelines http://www.mono-project.com/community/contributing/coding-guidelines/ but don't worry if you get something wrong. One of the project members will help you to get things landed.

Does your pull request fix any of the existing issues? Please use the following format: Fixes #issue-number
-->

CoffeeFlux reviewed (approved):
Assuming you'll push the required fixes once the original PR is ready. I can't test it anyway, so ¯\\\_(ツ)\_/¯
